tests/codegen-units/item-collection/instantiation-through-vtable.rs

```diff
@@ -4,6 +4,8 @@
 #![deny(dead_code)]
 #![feature(start)]
 
+#![allow(dead_code)]
+
```

Comment:
Please remove all the `allow(dead_code)` you added in tests.
Ui test already implicitly allow dead code, so this attribute is either useless, or breaks the test.
For instance, this test explicitly denies dead code, and should keep doing it.

Reply:
Oh, this is a mistake for tests with `#![deny(dead_code)]`, I will remove all `allow(dead_code)` in tests with `#![allow(dead_code)]`. But I really meet some tests which mark `// run-pass`, and then failed with new warning, so I add `#![allow(dead_code)]` for these ones. Should I keep them or remove also? I have no idea how to make tests success if remove `allow(dead_code)` in such tests. For example, for test https://github.com/rust-lang/rust/pull/118257/files#diff-7c21885e9503e8a512ff2e867e90613c4a5da1cf3e613269df0683259ddc1161, should I add all the expected WARN for dead code?

Reply:
> should I add all the expected WARN for dead code?

Yes. And this allows to verify that your pass does nothing wrong.